distributed/worker.py

```diff
@@ -91,6 +91,73 @@
 SerializedTask = namedtuple("SerializedTask", ["function", "args", "kwargs", "task"])
 
 
+class TaskState:
+    """Holds volatile state relating to an individual Dask task
+
+
+    * **dependencies**: ``set(TaskState instances)``
+        The data needed by this key to run
+    * **dependents**: ``set(TaskState instances)``
+        The keys that use this dependency
+    * **duration**: ``float``
+        Expected duration the a task
+    * **priority**: ``tuple``
+        The priority this task given by the scheduler.  Determines run order.
+    * **state**: ``str``
+        The current state of the task. One of ["waiting", "ready", "executing",
+        "memory", "flight", "executing", "error", "long-running",
+        "rescheduled", "error"]
+    * **who_has**: ``set(worker)``
+        Workers that we believe have this data
+    * **waiting_for_data**: ``set(keys of dependencies)``
+        A dynamic verion of dependencies.  All dependencies that we still don't
+        have for a particular key.
+    * **resource_restrictions**: ``{str: number}``
+        Abstract resources required to run a task
+    * **exception**: ``str``
+        The exception caused by running a task if it erred
+    * **traceback**: ``str``
+        The exception caused by running a task if it erred
+    * **type_**: ``type``
```

Comment:
Yeah, I didn't want to use the name `type`, even as an attribute.  Happy to change this if it reads weird.

Reply:
Gotcha. FWIW it reads a little weird to me, so I have a slight preference for switching to `type` instead of `type_` (similar to what we have in the [`FutureState` class](https://github.com/dask/distributed/blob/946c6a46abb04615051537f1bf39b81163aa91f0/distributed/client.py#L432))